pinot-common/src/main/java/org/apache/pinot/common/metadata/segment/SegmentZKMetadata.java

```diff
@@ -66,8 +66,11 @@ public String getSegmentName() {
   public long getStartTimeMs() {
     if (!_startTimeMsCached) {
       String startTimeString = _simpleFields.get(Segment.START_TIME);
-      if (startTimeString != null && Long.parseLong(startTimeString) > 0) {
-        _startTimeMs = TimeUnit.valueOf(_simpleFields.get(Segment.TIME_UNIT)).toMillis(Long.parseLong(startTimeString));
+      // NOTE: Need to check whether the start time is positive because some old segment ZK metadata contains negative
+      //       start time and null time unit
+      long startTime;
+      if (startTimeString != null && (startTime = Long.parseLong(startTimeString)) > 0) {
```

Comment:
It seems the build isn't happy about inner assignment. Maybe adding a private method to extract the field helps?